src/borg/locking.py

```diff
@@ -158,6 +190,47 @@ def is_locked(self):
     def by_me(self):
         return os.path.exists(self.unique_name)
 
+    def kill_stale_lock(self):
+        for name in os.listdir(self.path):
```

Comment:
somehow this code seems to be undecided whether it should deal with multiple filenames encountered in self.path or not.

you use "for" (== would be able to deal with multiple files), OTOH, you often just "return", so it stops processing.